client/allocrunner/taskrunner/script_check_hook.go

```diff
@@ -366,6 +370,11 @@ func newScriptCheckCallback(s *scriptCheck) taskletCallback {
 			outputMsg = string(output)
 		}
 
+		// If the check is unhealthy, log the output
+		if state == api.HealthCritical || state == api.HealthWarning {
+			s.logger.Warn("unhealthy script check", "health", state, "output", strconv.Quote(outputMsg))
```

Comment:
If the output message lands in Consul anyways, maybe we'd get most of the benefit of this if we logged the script check's error code rather than textual output?

Reply:
How about logging just the exist code by default, then opt-into full text output by setting DEBUG or TRACE mode?

Reply:
I think the problem is that the output in Consul is ephemeral. Consul does not log the output, so if you're not observing Consul at the time the failure happens, the output is gone forever.

We could make this Consul's problem I suppose.

Reply:
> We could make this Consul's problem I suppose.

Checked with Consul and @mkeeler felt it would be best to do this in Nomad since we're the one running the script. Nomad can also annotate the log line with more metadata than Consul has available (eg Alloc ID). Lacking the alloc id in the log line would really complicate using this to create a timeline of events after an outage.

Reply:
That makes sense. I don't _love_ that the allocation can write arbitrary strings to the client log but I'm sure if we looked hard enough we could find other cases of that.